illumos-utils/src/link.rs

```diff
@@ -178,29 +207,42 @@ impl Link {
 
 impl Drop for Link {
     fn drop(&mut self) {
-        let r = self.delete();
-        if let Err(e) = r {
-            eprintln!("Failed to delete Link: {}", e);
+        if let Some(destructor) = self.destructor.take() {
+            destructor
+                .enqueue_destroy(LinkDestruction { name: self.name.clone() });
         }
     }
 }
 
+// Represents the request to destroy a VNIC
+struct LinkDestruction {
```

Comment:
Not sure if we'll delete other kinds of links, but if so, consider renaming to `VnicDestruction`

Reply:
Done